### PR TITLE
[SCHEMA] Check sidecar fields mentioned in rule checks exist in the schema 

### PR DESCRIPTION
**Pull request by Remi-Gau**
- closes none but relates to https://github.com/bids-standard/bids-specification/pull/1910#issuecomment-2312591610

### CODE REVIEW

File: tools/schemacode/bidsschematools/tests/test_expressions.py [hunk 1]
```diff
@@ -77,3 +86,67 @@ def test_checks(schema_obj):
 def test_expected_failures(expr):
     with pytest.raises(ParseException):
         expression.parse_string(expr)
+
+
+def test_valid_sidecar_field(schema_obj):
```
Comment (Remi-Gau) [inline tools/schemacode/bidsschematools/tests/test_expressions.py:91]: I sort of totally improvised testing this, so there may be a simpler way to do this.

**Comment by Remi-Gau** [comment]
FYI this actually caught a few typos or things that had not been updated in the schema.

File: tools/schemacode/bidsschematools/tests/test_expressions.py [hunk 2]
```diff
@@ -77,3 +91,95 @@ def test_checks(schema_obj):
...
+                    assert (
+                        name.split(".", 1)[1] in field_names
+                    ), f"Bad field in selector: {name} ({key})"
+        for selector in rule.get("checks", []):
+            ast = expression.parse_string(selector)[0]
+            for name in find_names(ast):
+                if name.startswith(("json.", "sidecar.")):
+                    assert (
+                        name.split(".", 1)[1] in field_names
+                    ), f"Bad field in selector: {name} ({key})"
```
Comment (effigies) [inline tools/schemacode/bidsschematools/tests/test_expressions.py:128]:
```suggestion
                    ), f"Bad field in check: {name} ({key})"
```

File: tools/schemacode/bidsschematools/tests/test_expressions.py [hunk 3]
```diff
@@ -77,3 +91,95 @@ def test_checks(schema_obj):
...
+    ):
+        for selector in rule["selectors"]:
+            ast = expression.parse_string(selector)[0]
+            for name in find_names(ast):
+                if name.startswith(("json.", "sidecar.")):
+                    assert (
+                        name.split(".", 1)[1] in field_names
+                    ), f"Bad field in selector: {name} ({key})"
+        for selector in rule.get("checks", []):
+            ast = expression.parse_string(selector)[0]
```
Comment (effigies) [inline tools/schemacode/bidsschematools/tests/test_expressions.py:123]:
```suggestion
        for check in rule.get("checks", []):
            ast = expression.parse_string(check)[0]
```

**Review by effigies — APPROVED** [review]
Largely rewrote to fetch any identifier, whether it's a bare (`suffix`), a function name (`exists`) or a property (`nifti_header.dims`). This way we can add more checks easily at the top level.

I'm not going to bog this PR down with other checks, but, for example, we could make sure that `nifti_header` fields exist in `schema.meta.context.properties.nifti_header.properties`, or `schema.*` can be found in the schema itself.

I didn't find any new typos from what you already found by expanding to all rules that include selectors.

**Comment by Remi-Gau** [comment]
Thanks for the review and expansion.
At least this provides a framework to expand on.

**Pull request merged**